### PR TITLE
Add SubmitAction component

### LINKED ISSUE #33: Add submit button component
_As a user of the design system
I want to see the convention for submit buttons
and be provided with a reusable component
So that my service can be consistent with all other Barnardo's digital services_

## Description

Now that the design system includes rules and guidance on styles such as layout, typography and colour, it now needs to be expanded to include the reusable components that use these styles. 

The benefit of sharing reusable components is that it maximises efficiencies in product teams, reduces debate and duplication, and maintains consistency across all products and services.

The submit button is a logical place to start as there are, despite seeming simple, so many usability and accessibility considerations. 

## Acceptance criteria

- [x] Add a section with for the submit button component to the design system
- [x] Addresses accessibility/usability considerations
- [x] Works on target devices and browsers

### PR DESCRIPTION
**Pull request by kelliewellingtons**
Includes rules based on accessibility and usability research.

Example implementation includes hover, focus and active states.

Closes #33

### CODE REVIEW

**Comment by sl33p** [comment]
Is there no netliffffffy preview for this one?

**Comment by jeddy3** [comment]
It looks like the GitHub integration hiccuped. 

Deploy preview at:
https://5a981cd8df99530e6f708fb9--barnardos-design-system.netlify.com/

**Comment by jeddy3** [comment]
> All looking good! Do we want to provide different options for the markup? So having a button as well as a input?

At this stage, the code is just the minimum amount needed to _build out the design system itself_. We feel communicating, through the lists and examples, the usability and accessibility rules are the first priority.

We intend to tackle the actual implementation details (and features) of components at a later date.

File: src/components/SubmitAction/index.css [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  color: #ffffff;
+  font: 600 1.25rem/1.5rem Nova, sans-serif;
+  padding: 0.75rem;
+  text-align: center;
+  white-space: normal;
+  width: 100%;
+}
+
+.SubmitAction:hover {
+  background-color: color(#81bc00 shade(10%));
```
Comment (joelcarr) [inline src/components/SubmitAction/index.css]: Not seen `color(red shade(10%))` before. How is the support?

Comment (jeddy3) [reply src/components/SubmitAction/index.css]: Gatsby bundles [postcss-cssnext](http://cssnext.io/) by default. We're currently using it to transpile the `system-ui` generic font keyword. 

It [can also transpile](http://cssnext.io/features/#color-function) the `color()` function, an upcoming feature of the CSS langauge.

**Comment by sl33p** [comment]
👍✨

When the snow has melted we should spend 30 minutes discussing how accessible the explanatory content is (https://github.com/barnardos/design-system/issues/36), there has been a lot of thinking in condensing the principles into one liners, and I feel that some of that needs to go into an deeper explanation into the reasoning. 

- what benefit does the principle provide?
- is there research or resources backing up the principle?
- is there an outline of risks of not using the principle (as proving a negative is often easier)?

**Pull request merged**